src/services/__tests__/mail.test.js

```diff
@@ -0,0 +1,103 @@
+const nodemailer = require('nodemailer');
+const { sendEmail } = require('../mail');
+
+const spyOnCreateTransport = jest.spyOn(nodemailer, 'createTransport');
+
+afterAll(() => {
+  spyOnCreateTransport.mockRestore();
+});
+describe('sendEmail service function', () => {
+  test('Should send an email to all selected receivers', async () => {
+    const emailOptions = {
+      from: 'paying it forward',
+      to: 'test1@mail.com, test2@mail.com',
+      subject: 'test',
+    };
+    const sendMailResult = {
+      messageId: '<messageId>',
+      accepted: ['test1@mail.com, test2@mail.com'],
+      rejected: [],
+      envelopeTime: 0,
+      messageTime: 0,
+      response: '250 2.0.0 OK 1575454969 qp 1469',
+    };
+    const spyOnSendMail = jest.fn().mockResolvedValue(sendMailResult);
+    spyOnCreateTransport.mockReturnValue({
+      sendMail: spyOnSendMail,
+    });
+
+    const res = await sendEmail(emailOptions);
+    expect(spyOnCreateTransport).toHaveBeenCalled();
+    expect(spyOnCreateTransport).toHaveBeenCalledTimes(1);
+    expect(spyOnCreateTransport).toHaveReturned();
+    expect(spyOnCreateTransport).toHaveBeenCalledWith({
+      host: 'smtp.office365.com',
+      auth: {
+        user: process.env.EMAIL_USER,
+        pass: process.env.EMAIL_PASSWORD,
+      },
+    });
+    expect(spyOnSendMail).toHaveBeenCalled();
+    expect(spyOnSendMail).toHaveBeenCalledTimes(1);
+    expect(spyOnSendMail).toHaveBeenCalledWith(emailOptions);
+    expect(spyOnSendMail).not.toThrow();
+    expect(res).toEqual(sendMailResult);
+    spyOnSendMail.mockRestore();
+  });
+
+  test('Should throw an error if the there are no recipients', async () => {
+    jest.clearAllMocks();
+    const emailOptions = {
+      from: 'paying it forward',
+      subject: '',
+      html: '',
+    };
+    const spyOnSendMail = jest
+      .fn()
+      .mockRejectedValue(new Error('No recipients'));
+    spyOnCreateTransport.mockReturnValue({
+      sendMail: spyOnSendMail,
+    });
+
+    await sendEmail(emailOptions);
+    expect(spyOnCreateTransport).toHaveBeenCalled();
+    expect(spyOnCreateTransport).toHaveBeenCalledTimes(1);
+    expect(spyOnCreateTransport).toHaveReturned();
+    expect(spyOnSendMail).toHaveBeenCalled();
+    expect(spyOnSendMail).toHaveBeenCalledWith(emailOptions);
+    expect(spyOnSendMail).rejects.toThrow('No recipients');
+    spyOnSendMail.mockRestore();
+  });
+
+  // NOTE: I think this test is not necessary because invalid emails should be detected when user signs up
```

Comment:
I aggree with you on this case. However, there may be a case where an email address is actually pass the validity checks and still it is not valid. It may be deleted, suspended or somehow disabled before signing up or maybe after a while they signed up. So, maybe we can add a general test case for sending email failed.

Reply:
I agree with @mehmetfatiherdem 
This is a unit test for the `sendMail` function, this unit test does not know if another functionality with email validation exists. So it should check for all possible edge cases without assuming it's being tested somewhere else.

We can keep this test case as it, and you can remove the above comment.